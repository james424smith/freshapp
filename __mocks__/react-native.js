import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
