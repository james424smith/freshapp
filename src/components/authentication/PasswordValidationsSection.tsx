import React, { useState, useEffect } from "react";
import Text from "../StyledText";
import ActiveTick from "../../../assets/icons/tick_enable.svg";
import DisableTick from "../../../assets/icons/tick_disable.svg";
import styles from "../../styles/AuthStyles";
import { View } from "react-native";
interface Props {
  password: string;
  setAllValid: (arg: boolean) => void;
}

const PasswordValidationsSection = (props: Props) => {
  const { password, setAllValid } = props;
  const [
    containsEightCharacters,
    setContainsEightCharacters,
  ] = useState<boolean>(false); // min 8 characters
  const [containsUpperCase, setContainsUpperCase] = useState<boolean>(false); // upper character
  const [
    containsSpecialCharacter,
    setContainsSpecialCharacter,
  ] = useState<boolean>(false); // special character

  const mustContainData = [
    "At least 8 characters)",
    "Contains a capital letter",
    "Contains a symbol (!, @, #, $, %, ^, &, *)",
  ];

  useEffect(() => {
    const temp = validatePassword();

    if (temp.eightCharacters && temp.specialCharacter && temp.upperCase) {
      setAllValid(true);
    } else {
      setAllValid(false);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [password]);

  const ValidateEightCharacters = () => {
    if (password.length >= 8) {
      return true;
    } else {
      return false;
    }
  };
  const ValidateUpperCase = () => {
    if (password.toLowerCase() !== password) {
      return true;
    } else {
      return false;
    }
  };
  const ValidateSpecialCharacter = () => {
    if (/[!,@,#,$,%,^,&,*]/g.test(password)) {
      return true;
    } else {
      return false;
    }
  };

  const validatePassword = () => {
    // has 8 characters
    const eightCharacters = ValidateEightCharacters();
    setContainsEightCharacters(eightCharacters);
    const upperCase = ValidateUpperCase();
    setContainsUpperCase(upperCase);
    const specialCharacter = ValidateSpecialCharacter();
    setContainsSpecialCharacter(specialCharacter);
    return { eightCharacters, upperCase, specialCharacter };
  };

  return (
    <View style={styles.passwordValidationSection}>
      {[
        containsEightCharacters,
        containsUpperCase,
        containsSpecialCharacter,
      ].map((isTrue, i) => (
        <View key={i} style={styles.passwordValidationMapChecks}>
          <View style={styles.passwordValidationCheckIcon}>
            {isTrue ? (
              <ActiveTick
                width={18}
                height={18}
                options={{ alignSelf: "center" }}
              />
            ) : (
              <DisableTick width={18} height={18} />
            )}
          </View>
          <View>
            <Text
              style={
                isTrue
                  ? styles.passwordValidationCheckedIconText
                  : styles.passwordValidationUncheckedIconText
              }
            >
              {mustContainData[i]}
            </Text>
          </View>
        </View>
      ))}
    </View>
  );
};

export default PasswordValidationsSection;
