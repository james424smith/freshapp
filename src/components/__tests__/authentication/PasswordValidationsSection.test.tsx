import React from "react";
import PasswordValidationsSection from "../../authentication/PasswordValidationsSection";
import { render, cleanup } from "@testing-library/react-native";

describe("PasswordValidationsSection snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render correct ui when password matches the criterias", () => {
    const func = jest.fn();

    const { toJSON } = render(
      <PasswordValidationsSection
        password={"SomePassword543!"}
        setAllValid={func}
      />
    );

    expect(func).toHaveBeenCalledWith(true);
    expect(toJSON()).toMatchSnapshot();
  });
  it("should render correct ui when password doesn't match the criterias", () => {
    const func = jest.fn();

    const { toJSON } = render(
      <PasswordValidationsSection password={"123"} setAllValid={func} />
    );

    expect(func).toHaveBeenCalledWith(false);
    expect(toJSON()).toMatchSnapshot();
  });
});
