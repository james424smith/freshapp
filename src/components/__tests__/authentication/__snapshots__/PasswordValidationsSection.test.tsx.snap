// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordValidationsSection snapshot test should render correct ui when password doesn't match the criterias 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "width": 75,
        }
      }
    >
      <<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><defs><style>.a{fill:#e0e0e0;}.b{fill:#fff;}</style></defs><g transform="translate(0)"><circle class="a" cx="8" cy="8" r="8" transform="translate(0 0)"/><path class="b" d="M79.374,110.145,73.718,115.8l-2.775-2.775-1.409,1.409,4.184,4.184,7.064-7.065Z" transform="translate(-67.158 -106.382)"/></g></svg>
        height={18}
        width={18}
      />
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignSelf": "center",
              "color": "#78849E",
              "width": 525,
            },
          ]
        }
      >
        At least 8 characters)
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "width": 75,
        }
      }
    >
      <<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><defs><style>.a{fill:#e0e0e0;}.b{fill:#fff;}</style></defs><g transform="translate(0)"><circle class="a" cx="8" cy="8" r="8" transform="translate(0 0)"/><path class="b" d="M79.374,110.145,73.718,115.8l-2.775-2.775-1.409,1.409,4.184,4.184,7.064-7.065Z" transform="translate(-67.158 -106.382)"/></g></svg>
        height={18}
        width={18}
      />
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignSelf": "center",
              "color": "#78849E",
              "width": 525,
            },
          ]
        }
      >
        Contains a capital letter
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "width": 75,
        }
      }
    >
      <<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><defs><style>.a{fill:#e0e0e0;}.b{fill:#fff;}</style></defs><g transform="translate(0)"><circle class="a" cx="8" cy="8" r="8" transform="translate(0 0)"/><path class="b" d="M79.374,110.145,73.718,115.8l-2.775-2.775-1.409,1.409,4.184,4.184,7.064-7.065Z" transform="translate(-67.158 -106.382)"/></g></svg>
        height={18}
        width={18}
      />
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignSelf": "center",
              "color": "#78849E",
              "width": 525,
            },
          ]
        }
      >
        Contains a symbol (!, @, #, $, %, ^, &, *)
      </Text>
    </View>
  </View>
</View>
`;

exports[`PasswordValidationsSection snapshot test should render correct ui when password matches the criterias 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "width": 75,
        }
      }
    >
      <<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><defs><style>.a{fill:#09b3a5;}.b{fill:#ebf0f3;}</style></defs><g transform="translate(0)"><circle class="a" cx="8" cy="8" r="8" transform="translate(0 0)"/><path class="b" d="M79.374,110.145,73.718,115.8l-2.775-2.775-1.409,1.409,4.184,4.184,7.064-7.065Z" transform="translate(-67.158 -106.382)"/></g></svg>
        height={18}
        options={
          Object {
            "alignSelf": "center",
          }
        }
        width={18}
      />
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignSelf": "center",
              "color": "#09B3A5",
              "width": 525,
            },
          ]
        }
      >
        At least 8 characters)
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "width": 75,
        }
      }
    >
      <<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><defs><style>.a{fill:#09b3a5;}.b{fill:#ebf0f3;}</style></defs><g transform="translate(0)"><circle class="a" cx="8" cy="8" r="8" transform="translate(0 0)"/><path class="b" d="M79.374,110.145,73.718,115.8l-2.775-2.775-1.409,1.409,4.184,4.184,7.064-7.065Z" transform="translate(-67.158 -106.382)"/></g></svg>
        height={18}
        options={
          Object {
            "alignSelf": "center",
          }
        }
        width={18}
      />
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignSelf": "center",
              "color": "#09B3A5",
              "width": 525,
            },
          ]
        }
      >
        Contains a capital letter
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "width": 75,
        }
      }
    >
      <<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><defs><style>.a{fill:#09b3a5;}.b{fill:#ebf0f3;}</style></defs><g transform="translate(0)"><circle class="a" cx="8" cy="8" r="8" transform="translate(0 0)"/><path class="b" d="M79.374,110.145,73.718,115.8l-2.775-2.775-1.409,1.409,4.184,4.184,7.064-7.065Z" transform="translate(-67.158 -106.382)"/></g></svg>
        height={18}
        options={
          Object {
            "alignSelf": "center",
          }
        }
        width={18}
      />
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignSelf": "center",
              "color": "#09B3A5",
              "width": 525,
            },
          ]
        }
      >
        Contains a symbol (!, @, #, $, %, ^, &, *)
      </Text>
    </View>
  </View>
</View>
`;
